### PR TITLE
Bugfix for xWebApplication

### PR DESCRIPTION
**Pull request by SNikalaichyk**
Bugfix for xWebApplication Set-TargetResource

There were several issues in the Set-TargetResource function of the xWebApplication DSC resource (MSFT_xWebApplication.psm1):

1. Line 94. $ApplicationPool must be replaced with $WebAppPool.

2. Line 96. Verbose message must be updated.

3. Line 92 and 97. It is strongly recommended to use cmdlet Set-WebConfigurationProperty instead of Set-ItemProperty because of the issue described below.

Issue description:
If site root directory contains a subfolder that has the same name as the Web Application, they both have identical PSPaths under "IIS:\Sites\<SiteName>".

The Set-ItemProperty cmdlet in the Set-TargetResource function occasionally selects the item of type "directory" instead of "application" and eventually fails with the following error:
"ErrorMessage is The PowerShell provider MSFT_xWebApplication threw one or more non-terminating errors while running the Set-TargetResource functionality. 
FullyQualifiedErrorId is InvalidArgument,Microsoft.PowerShell.Commands.SetItemPropertyCommand. ErrorMessage is Value cannot be null.
Parameter name: path."

Steps to reproduce the issue:

PS IIS:\> Get-ChildItem "IIS:\Sites\Default Web Site" | Where {$_.Name -eq "EQueueClient"} | Select NodeType, PSPath

NodeType : directory
PSPath   : WebAdministration::\\WebServer01\Sites\Default Web Site\EQueueClient

NodeType : application
PSPath   : WebAdministration::\\WebServer01\Sites\Default Web Site\EQueueClient

PS IIS:\> Set-ItemProperty "IIS:\Sites\Default Web Site\EQueueClient" -Name applicationPool -Value DefaultAppPool
Set-ItemProperty : Value cannot be null.
Parameter name: path
At line:1 char:1
+ Set-ItemProperty "IIS:\Sites\Default Web Site\EQueueClient" -Name applicationPoo ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Set-ItemProperty], ArgumentNullException
    + FullyQualifiedErrorId : InvalidArgument,Microsoft.PowerShell.Commands.SetItemPropertyCommand

### CODE REVIEW

**Comment by SNikalaichyk** [comment]
AppVeyor build failed because of tabs instead of spaces. Should I replace tabs with spaces and create a new pull request?

**Comment by tysonjhayes** [comment]
The change looks pretty good to me (once the tabs/spaces thing is corrected). However it would be nice to see some tests on this so we don't regress.

**Comment by SNikalaichyk** [comment]
I'm a newbie here on GitHub. I tested the change before submitting the pull request. What kind of evidence do you want me to provide (screenshots or something else)?

**Comment by KarolKaczmarek** [comment]
Thanks for contribution @SNikalaichyk! 
It looks good, but as @tysonjhayes mentioned, it would be good to add some test automation.

Tests are being run automatically by AppVeyor after sending the PR. You can see results at the bottom of the PR (green check with ```All checks have passed``` sentence), if you click on ```show all checks``` you can also go directly to AppVeyor summary of given test pass (after clicking on ```Details```). Click [here](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#appveyor) for more info on AppVeyor.

However, those are [common tests](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#common-tests)  and it would be nice to have additional ones specifically targeting your changes.

Please take a look at our contributing guide, especially [section about test coverage](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#improving-test-coverage-for-existing-resources) for more information on adding test automation. Instruction on how to run tests can be found [here](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#run-common-tests-locally).

Please let me know if you have further questions.

**Comment by tysonjhayes** [comment]
Hi @SNikalaichyk  - are you still working on this? It's been a while since we've seen any activity on this. If you are not working on this anymore (or I don't hear from you) I'm going to fork this and try to fix it up myself as we'd really like some of this functionality. Please let me know.

**Comment by SNikalaichyk** [comment]
Hi @tysonjhayes,
I could not figure out quickly how to test these particular changes with Pester, and completely forgot about this issue. Please give me a day or two.

**Comment by tysonjhayes** [comment]
No worries! If you need help please let me know I'm happy to assist!

File: Tests/Unit/MSFT_xWebApplication.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+InModuleScope -ModuleName  $DSCResourceName -ScriptBlock {
+
+    $MockParameters = @{
+        Website       = 'MockSite'
+        Name          = 'MockApp'
+        WebAppPool    = 'MockPool'
+        PhysicalPath  = 'C:\MockSite\MockApp'
+    }
+
+    Describe 'CheckDependencies' {
```
Comment (tysonjhayes) [inline Tests/Unit/MSFT_xWebApplication.Tests.ps1]: Could we change this to:
````powershell
Describe 'MSFT_xWebApplicaiton\CheckDependancies'
````

So we know which funciton we're calling where in our ever growing pile of tests?

File: Tests/Unit/MSFT_xWebApplication.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+                    CheckDependencies
+                } | Should Throw 'Please ensure that WebAdministration module is installed.'
+            }
+
+        }
+
+    }
+
+
+    Describe 'Get-TargetResource' {
```
Comment (tysonjhayes) [inline Tests/Unit/MSFT_xWebApplication.Tests.ps1]: Could we change this to:
````powershell
Describe 'MSFT_xWebApplicaiton\Get-TargetResource'
````
So we know which funciton we're calling where in our ever growing pile of tests?

File: Tests/Unit/MSFT_xWebApplication.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+                $Result = Get-TargetResource @MockParameters
+                $Result.Ensure | Should Be 'Present'
+            }
+
+        }
+
+    }
+
+
+    Describe "how Test-TargetResource responds to Ensure = 'Absent'" {
```
Comment (tysonjhayes) [inline Tests/Unit/MSFT_xWebApplication.Tests.ps1]: Could we change this to:
````powershell
Describe "how MSFT_xWebApplication\Test-TargetResource responds to Ensure = 'Absent'"
````
So we know which funciton we're calling where in our ever growing pile of tests?

File: Tests/Unit/MSFT_xWebApplication.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+                $Result = Test-TargetResource -Ensure 'Absent' @MockParameters
+                $Result | Should Be $false
+            }
+
+        }
+
+    }
+
+
+    Describe "how Test-TargetResource responds to Ensure = 'Present'" {
```
Comment (tysonjhayes) [inline Tests/Unit/MSFT_xWebApplication.Tests.ps1]: I'm sure you've gotten the idea at this point.

**Comment by tysonjhayes** [comment]
This looks great, I'd like to get a few minor changes in and we can get this merged.

**Comment by tysonjhayes** [comment]
Thanks for everything!

**Pull request merged**